src/test/java/org/mule/tools/apikit/MainAppScaffolderWithExistingConfigApiSyncTest.java

```diff
@@ -60,19 +100,71 @@ public void reScaffoldDifferentVersionsWithGlobal() throws Exception {
     ApiReference apiReference = ApiReference.create(RAML_RESOURCE_URL_V1, testScaffolderResourceLoader);
 
     ScaffoldingResult result = scaffoldApi(RuntimeEdition.CE, apiReference, null,
-                                           asList(existingConfigV1Folder + "/api.xml"));
+                                           createMuleConfigsFromLocations(asList(existingConfigV1Folder + "/api.xml")), null);
     verifySuccessfulScaffolding(result, existingConfigV1Folder + "/api.xml");
 
     testScaffolderResourceLoader = new TestScaffolderResourceLoader(existingConfigV2Folder);
     apiReference = ApiReference.create(RAML_RESOURCE_URL_V2, testScaffolderResourceLoader);
 
     List<String> muleConfigs = asList(existingConfigV1Folder + "/api_refactored.xml", existingConfigV1Folder + "/global.xml");
     ScaffoldingResult rescaffoldResult =
-        scaffoldApi(RuntimeEdition.CE, apiReference, null, muleConfigs);
+        scaffoldApi(RuntimeEdition.CE, apiReference, null, createMuleConfigsFromLocations(muleConfigs), null);
     verifySuccessfulScaffolding(rescaffoldResult, existingConfigV2Folder + "/api_refactored.xml",
                                 existingConfigV2Folder + "/global.xml");
   }
 
+  /**
+   * First it takes an API normally with configuration of APIKit, http and api autodiscovery inside global xml file.
+   * Finally it simulates re-scaffolding.
+   * Expected behaviour is that the new scaffolded API has two configurations, globals.xml(with common configurations) and api.xml.
+   */
+  @Test
+  public void reScaffoldWithGlobalsAndAPIAutodiscovery() throws Exception {
+    String existingConfigFolder = TEST_RESOURCES_APISYNC_W_GLOBAL + "/api-autodiscovery";
+
+    ResourceLoader testScaffolderResourceLoader = new TestScaffolderResourceLoader(existingConfigFolder);
+    ApiReference apiReference = ApiReference.create(RAML_RESOURCE_URL_V1, testScaffolderResourceLoader);
```

Comment:
What is the purpose of this? This is never used because you are overriding the reference below.